### PR TITLE
AMDGPU: Increase the LDS size to support to 160 KB for gfx950

### PR DESCRIPTION
**Pull request by arsenm**

### CODE REVIEW

**Comment by arsenm** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/116309?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#116309** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/116309?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/116309?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#116308** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/116308?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#116307** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/116307?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

**Review by shiltian — APPROVED** [review]

File: llvm/lib/Target/AMDGPU/AMDGPU.td [hunk 1]
```diff
@@ -1494,7 +1494,8 @@ def FeatureISAVersion9_5_Common : FeatureSet<
   [FeatureFP8Insts,
    FeatureFP8ConversionInsts,
    FeatureCvtFP8VOP1Bug,
-   FeatureGFX950Insts
+   FeatureGFX950Insts,
+   FeatureAddressableLocalMemorySize163840
```
Comment (jayfoad) [inline llvm/lib/Target/AMDGPU/AMDGPU.td:1498]: This means that gfx950 has both `FeatureAddressableLocalMemorySize65536` and `FeatureAddressableLocalMemorySize163840`, which seems like it should be disallowed in `GCNSubtarget::checkSubtargetFeatures`.

Comment (arsenm) [reply llvm/lib/Target/AMDGPU/AMDGPU.td:1498]: It does end up taking the maximum, so it works

**Comment by arsenm** [comment]
### Merge activity

* **Nov 18, 1:34 PM EST**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/llvm/llvm-project/116309).

**Pull request merged**